src/Views/VPNPage.vala

```diff
@@ -246,7 +250,7 @@ public class Network.VPNPage : Network.Widgets.Page {
                 return;
             } else {
                 remove_vpn_toast.send_notification ();
-                sel_row.hide ();
+                vpn_list.remove (sel_row);
```

Comment:
This breaks the `delete_connection ()` method's check for `if (selected_row != null && sel_row != null)` as `selected_row` becomes `null`. and `selected_row.connection.delete (null)` is never actually called. That is why I used hide and show instead of remove and add. I used `selected_row` here since a user may change the selection while operation is running and update row in `sel_row`.